tools/autograd/gen_inplace_or_view_type.py

```diff
@@ -276,6 +280,9 @@ def unpack_args(f: NativeFunction) -> Tuple[List[str], List[Binding]]:
                 ref="&" if ref else "",
             )
         )
+        # This is broken because I am passing in "reference" tensor to my view op
+        # This gets turned into "reference_" in emit_view_lambda.
+        # Couldn't find the right if condition to only target this new CTyp inclusion
```

Comment:
For experimenting I just altered the AdInplaceOrView codegen file to remove the trailing underscore

Reply:
Added a fix to the codegen "reference" not sure if it is the right fix. Also ideally we wouldnt' save the whole tensor for the backward since we only need the metadata to for backward pass. HOWEVER this function is only needed "currently"(I am sure someone will want to use it else where) to generate the derivative of `_to_buffer`. Maybe the lifetime of the "reference" is guaranteed to be needed so it is really keeping the tensor around longer than needed but not sure this is true/not as elegant as just storing the metadata.

Reply:
It is NOT ok to save Tensors in this code!
As discussed above, I think this argument should most likely go away anyways and size/stride/offset should be passed in explicitly.